src/transformations/utils/build-super-calls-for-chain.ts

```diff
@@ -83,19 +81,46 @@ export function buildSuperCallsForChain(
   // once we have gathered all constructor calls for each parent, we linearize
   // them according to chain.
   const linearizedCtorCalls: string[] = [];
+  // this is where we store the parents of uninitialized parents if any
+  const invalidParents: number[] = [];
+  // Remove uninitialized parents's parents from linearization, and erase if they already are linearized
+  chain.forEach(parentNode => {
+    if (parentNode !== contractNode) {
+      // step 1 check if initializable
+      const args = ctorCalls[parentNode.id]?.call?.arguments;
+
+      if (
+        args === undefined &&
+        (getInitializerItems(parentNode).empty || !isImplicitlyConstructed(parentNode))
```

Comment:
I don't think this logic is right. If a constructor is empty, it shouldn't imply that we ignore all of its parents. For example:

```solidity
contract A {
  uint a;
  constructor() {
    a = 1;
  }
}

contract B is A {
  constructor() {}
}

contract C is B {
}
```

We would want `C_init` to invoke `A_init_unchained` in this scenario and I believe this logic would remove it. Please add this test case.

An important thing to understand if that `isImplicitlyConstructed` implies that we can default to `args = []`.

So we could define `const initializable = args !== undefined || isImplicitlyConstructed`. A parent is initializable if we either know the args, or we know that there are zero args i.e. we can default to `[]`.

Then the condition for removing all parents would simply be `if (!initializable)`.

Whether a parent constructor is `empty` is not relevant to marking parents as invalid. It should only be relevant in the next `for` loop: if a parent constructor is empty we can optimize by not printing the unchained init call, but its parents may still be relevant (as shown in the example above).

Some of the terms here are leading to confusion. I'd suggest renaming `empty` -> `emptyUnchained`, and `invalidParents` -> `notInitializable`.

Reply:
I see what you mean, thanks for bringing it up